### PR TITLE
Fix GDScriptCompiler not handling meta-types correctly

### PR DESCRIPTION
**Pull request by anvilfolk**
Some recent changes made it so that GDScript enum types were considered constant. However, the compiler didn't know how to handle those. This fixes that. Simple fix for issues mentioned in comments of #70495.

Should get rid of these:

```ERROR: Condition "!method" is true. Returning: false
   at: is_builtin_method_vararg (core/variant/variant_call.cpp:1339)
ERROR: Condition "!method" is true. Returning: 0
   at: get_builtin_method_argument_count (core/variant/variant_call.cpp:1265)
ERROR: Condition "!method" is true. Returning: Variant::NIL
   at: get_builtin_method_return_type (core/variant/variant_call.cpp:1318)
ERROR: Condition "!method" is true. Returning: nullptr
   at: get_validated_builtin_method (core/variant/variant_call.cpp:1251)
```

### CODE REVIEW

**Comment by anvilfolk** [comment]
Added an MRP not requiring two files. Thanks @dalexeev for the reminder to do this :)

**Comment by anvilfolk** [comment]
@adamscott I removed the `if (p_base != nullptr && p_base->is_constant) {` surrounding the `ENUM` check in `reduce_identifier_from_base` that you added in #70220 since with this little fix I found no regressions. I was wondering if I'm not creating issues down the line - what was the reason that was there in the first place?

I figure an `ENUM` is _always_ a constant, no matter what. the only other place that has that check is for classes, but without digging a lot further I'm not super certain of what those checks are doing.

File: modules/gdscript/tests/scripts/analyzer/features/enum_dictionary_func.gd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Thanks to dalexeev for the MRP!
+
```
Comment (dalexeev) [inline modules/gdscript/tests/scripts/analyzer/features/enum_dictionary_func.gd:2]:
```suggestion
```

Comment (anvilfolk) [reply modules/gdscript/tests/scripts/analyzer/features/enum_dictionary_func.gd:2]: Nooooooooooooooooooooooooooooooooooo but recognition is important :'(

If you really don't want it I'll take it off though :)

Comment (dalexeev) [reply modules/gdscript/tests/scripts/analyzer/features/enum_dictionary_func.gd:2]: I really think that the source code is not the right place for thanks, only useful information should be in the comments. However, thank you for your attention and good intentions. It made me smile as yesterday was my birthday.

Comment (anvilfolk) [reply modules/gdscript/tests/scripts/analyzer/features/enum_dictionary_func.gd]: I am so glad! Happy birthday! :) I have removed the comment - hope you have a lovely day!

**Comment by vonagam** [comment]
Some problems still remain:
```gdscript
enum MyEnum {}
# const MyConst := MyEnum

func _ready():
  const MyConst := MyEnum

  print(MyEnum.size()) # 0
  print(MyConst.size()) # runtime error about invalid call on base 'Nil'
```
If `MyConst` is a member and not local (like in comment) then the program has other pre-runtime errors.

So there are 3 other relevant places where `reduced_value` is used in compiler that need to be update:
1) [Member constants.](https://github.com/godotengine/godot/blob/b6e06038f8a373f7fb8d26e92d5f06887e459598/modules/gdscript/gdscript_compiler.cpp#L2409)
2) [Member variable.](https://github.com/godotengine/godot/blob/b6e06038f8a373f7fb8d26e92d5f06887e459598/modules/gdscript/gdscript_compiler.cpp#L2396)
3) [Local constant.](https://github.com/godotengine/godot/blob/b6e06038f8a373f7fb8d26e92d5f06887e459598/modules/gdscript/gdscript_compiler.cpp#L1956)

**Comment by anvilfolk** [comment]
Seems related to #54018.

I'll see whether I can dig into this some more, but maybe it'd be nice to prioritize getting rid of the `!method`

**Comment by anvilfolk** [comment]
Superseded by #70613! Thanks @vonagam for the lovely idea in there!

**Pull request closed without merging**